src/examples/crypto/rsa/rsa.ts

```diff
@@ -97,11 +101,7 @@ function multiply(
   for (let i = 0; i < 2 * 18 - 2; i++) {
     let res_i = res[i].add(carry);
 
-    [carry] = Provable.witnessFields(1, () => {
-      let res_in = res_i.toBigInt();
-      if (res_in > 1n << 128n) res_in -= Field.ORDER;
-      return [res_in >> 116n];
-    });
+    carry = res_i.div(2n ** 116n);
```

Comment:
awesome fix! since this already proves ` res_i.assertEquals(carry.mul(1n << 116n));`, we can remove the line below

Reply:
`.div(2n ** 116n)` solved the bug it added a ton of constraints.

I fixed the problem again using 
````typescript
carry = Provable.witness(Field, () => {
     let res_in = res_i.toBigInt();
     return Field(res_in * (Fp.inverse(2n ** 116n) ?? 0n));
});
````